src/relay/blockchain/events_informations.py

```diff
@@ -343,29 +346,67 @@ def get_transfer_path(sorted_balance_updates):
     return path_from_events
 
 
-def get_balance_update_events_for_transfer(all_events, transfer_event):
+def find_transfer_event(
+    all_events: Iterable[BlockchainEvent], event: BlockchainEvent
+) -> TransferEvent:
+    """Finds the corresponding transfer event, where event can by any event of the transfer
+    (BalanceUpdate or Transfer)
+    """
+
+    log_index = event.log_index
+    # Search forward for related Transfer update event
+    sorted_events = list(
+        sorted(
+            [event for event in all_events if event.log_index >= log_index],
+            key=attrgetter("log_index"),
+        )
+    )
+    next_log_index = log_index
+    for event in sorted_events:
+        if event.log_index == next_log_index:
+            if event.type == TransferEventType:
+                assert isinstance(event, TransferEvent)
+                return event
+
+            assert (
+                event.type == BalanceUpdateEventType
+            ), "Wrong event type for events before transfer event"
+
+            next_log_index = next_log_index + 1
+        else:
```

Comment:
it happens if there is hole in there, so if log_indexes are 1,2,3,5 for example, because there was a Trustline Update in between, so it means what we got was not a BalanceUpdateEvent of a Transfer but of a TrustlineUpdate

Reply:
But if it is not clear enough I need to change it maybe to make it more obvious how it works.

Reply:
I understand it now, but I think you could write a comment on how events are emitted from the contracts (and in what order) because it is the knowledge we need to understand it.

Reply:
done